packages/trader/src/AppV2/Hooks/useActiveSymbols.ts

```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect, useCallback, useRef } from 'react';
+import { WS, pickDefaultSymbol } from '@deriv/shared';
+import { useStore } from '@deriv/stores';
+import { ActiveSymbols } from '@deriv/api-types';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { usePrevious } from '@deriv/components';
+
+type TUseActiveSymbols = {
+    contract_type?: string[];
+    barrier_category?: string[];
+};
+//TODO: contract_type and barrier_category need to come from trade-store after calling contracts_for
+const useActiveSymbols = ({ contract_type = [], barrier_category = [] }: TUseActiveSymbols) => {
+    const [activeSymbols, setActiveSymbols] = useState<ActiveSymbols | []>([]);
+    const { client, modules } = useStore();
+    const { is_logged_in } = client;
+
+    const trader = useTraderStore();
+
+    const default_symbol_ref = useRef<string>('');
+    const previous_logged_in = usePrevious(is_logged_in);
+
+    const fetchActiveSymbols = useCallback(async () => {
+        let response;
+
+        const request = {
+            active_symbols: 'brief',
+            contract_type,
+            barrier_category,
+        };
+
+        if (is_logged_in) {
+            response = await WS.authorized.activeSymbols(request);
+        } else {
+            response = await WS.activeSymbols(request);
+        }
+
+        const { active_symbols, error } = response;
+
+        trader.active_symbols = active_symbols;
+
+        if (!active_symbols?.length || error) {
+            setActiveSymbols([]);
+        } else {
+            setActiveSymbols(active_symbols);
+            default_symbol_ref.current = trader.symbol || (await pickDefaultSymbol(active_symbols)) || '1HZ100V';
+            modules.trade.onChange({ target: { name: 'symbol', value: default_symbol_ref.current } });
+        }
+    }, [is_logged_in, trader, contract_type, barrier_category]);
+
+    useEffect(() => {
+        const is_logged_in_changed = previous_logged_in != undefined && previous_logged_in != is_logged_in;
+        if (trader.active_symbols?.length && !is_logged_in_changed) {
+            setActiveSymbols(trader.active_symbols);
```

Comment:
Do we need to call the `setActiveSymbols` again? It's already set in line 45.

![image](https://github.com/user-attachments/assets/cfc9d448-8199-4ec0-aef0-28d106c31478)

Reply:
yes i believe so. Active symbols from line 45 comes from api response. Where as here it comes from trade store for cases where we avoid calling active_symbols everytime this hook is used.

Reply:
For initialization, line 43 is not needed since `activeSymbols` are [] initially.
For a case with an API error: Do we want to remove them when there is some kind of an API error?  if we still wanna show `activeSymbols` even if there's an API error, then we don't need to reset them to [] at line 43.
If you have line 43 removed, `activeSymbols` will be set to either [] or `active_symbols`, whatever `active_symbols` contain, at line 45 :)